test/code/plugins/out_oms_diag_plugintest.rb

```diff
@@ -0,0 +1,116 @@
+require 'fluent/test'
+require_relative '../../../source/code/plugins/oms_configuration'
+require_relative '../../../source/code/plugins/oms_diag'
+require_relative '../../../source/code/plugins/out_oms_diag'
+
+class OutOMSDiagPluginTest < Test::Unit::TestCase
+    class Fluent::OutputOMSDiag
+
+        attr_reader :mock_data
+
+        def initialize
+            super
+            @mock_data = {}
+        end
+
+        def handle_record(ipname, record)
+            @mock_data[ipname] = record
+        end
+    end
+
+    class OMS::Configuration
+        class << self
+            def configurationLoaded=(configLoaded)
+                @@ConfigurationLoaded = configLoaded
+            end
+        end
+    end
+
+    def setup
+        Fluent::Test.setup
+        # mock the configuration to be loaded
+        OMS::Configuration.configurationLoaded = true
+    end
+
+    def teardown
+    end
+
+    def create_spurious_dataitem(log, ipname)
+        dataitem = create_valid_dataitem(log, ipname)
+        # only delete logmessage or time as ipname is set to default if absent
+        # thereby having no ipname before sending to write is non spurious
```

Comment:
Check the comment above on line 159 of oms_diag_test.rb to understand this comment in source code.